src/main/resources/templates/go/solution/bubblesort.go

```diff
@@ -0,0 +1,19 @@
+package ${packageName}
+
+import "time"
+
+type BubbleSort struct{}
+
+func NewBubbleSort() *BubbleSort {
+	return new(BubbleSort)
+}
+
+func (b BubbleSort) PerformSort(input []time.Time) {
```

Comment:
In this case it doesn't really matter as you do not access/modify `b` at all, but usually using `(b * Bubblesort)` as a receiver type would be better here. (Basically like you do it with Context)

 https://stackoverflow.com/questions/25382073/defining-golang-struct-function-using-pointer-or-not

Reply:
I agree, pointer receivers are more conventional.
Implemented in https://github.com/ls1intum/Artemis/pull/9751/commits/a7227c693eb841c7a9357f764829b325089a6f8d.